src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java

```diff
@@ -586,7 +586,15 @@ else if (participation.getExercise() instanceof ProgrammingExercise) {
         }
 
         User user = userService.getUserWithGroupsAndAuthorities();
-        checkAccessPermissionAtInstructor(participation, user);
+
+        // Allow USER's and TA's to delete their own StudentParticipations
+        if (user.getId().equals(participation.getStudent().getId())) {
+            checkAccessPermissionAtLeastStudent(participation, user);
```

Comment:
can we add an additional check for now, that this can only happen in tutorial exercises for normal users and TAs?
I think we store them in the yml file anyway, so we should be able to write a util function that checks this.

I imagine some logic like:

1) If isAtLeastInstructor(...) { //do normal delete }
2) else if isTA or isStudent(...) { // check if the partcipation belongs to a tutorial exercise }

Another (maybe cleaner) alternative would be to introduce a new API endpoint for the case of deleting participations for tutorial exercises.

Reply:
At first I have implemented a new API endpoint for deleting participations for tutorial exercises, but then I remembered that you once mentioned that it should be possible for tutors and student to delete their own participation in the future. As suggested by you, I will add another check logic here :)